src/main/java/academicweapon/Duke.java

```diff
@@ -0,0 +1,146 @@
+package academicweapon;
+
+import academicweapon.action.Action;
+import academicweapon.exceptions.DukeExceptions;
+import academicweapon.parser.Parser;
+import academicweapon.storage.Storage;
+import academicweapon.task.Task;
+import academicweapon.task.TaskList;
+import academicweapon.task.Deadline;
+import academicweapon.task.Event;
+import academicweapon.task.Todo;
+import academicweapon.ui.Ui;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+/**
+ * Duke is a task management application that allows users to interact with tasks through a command-line interface.
+ * It supports various command for managing tasks such as adding, listing, marking and deleting tasks.
+ */
+public class Duke {
+
+    private Storage storage;
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructs a Duke object with the specified file path for storage
+     *
+     * @param filePath The file path for task storage
+     */
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeExceptions e) {
+            ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    /**
+     * Runs the duke application, handling user interactions through a command-line interface.
+     * It processes user commands, updates tasks, and performs necessary  actions based on the commands.
+     */
+    public void run() {
+        ui.showWelcome();
+        boolean isExit = false;
+        while (!isExit) {
+            String fullCommand = "";
+            try {
+                fullCommand = ui.readCommand();
+            } catch (IOException e) {
+                System.out.println(e.getMessage());
+                continue;
+            } catch (DukeExceptions e) {
+                System.out.println(e.getMessage());
+                continue;
+            }
+
+            ui.showLine();
+            ArrayList<String> actions = new ArrayList<>();
+
+            try {
+                actions = Parser.parse(fullCommand);
+            } catch (DukeExceptions e) {
+                System.out.println(e.getMessage());
+                continue;
+            }
+
+            Action action = Action.valueOf(actions.get(0));
+
+            switch (action) {
+            case FIND:
+                ArrayList<String> strLst = this.tasks.findKeyword(actions.get(1));
```

Comment:
I think it would be nice to name strLst into something more direct like what the findKeyword function returns

Reply:
Collection names should also be in plural